Sources/TripKitUI/views/TKUIModePicker.swift

```diff
@@ -56,10 +43,62 @@ public class TKUIModePicker<Item>: UIView where Item: TKUIModePickerItem {
   private var visibleModes: [Item] = []
   
   public weak var containerView: UIView?
+  
+  private weak var collectionView: UICollectionView!
+  private weak var collectionViewHeightConstraint: NSLayoutConstraint!
+  private var layoutHelper: TKUIModePickerLayoutHelper?
 
   /// Whether user can enable/disable modes
   public var isEnabled: Bool = true
   
+  private var lastTappedItem: Item?
+  
+  public init() {
+    super.init(frame: .zero)
+    prepareCollectionView()
+  }
+  
+  required init?(coder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+  
+  private func prepareCollectionView() {
+    let collectionView = UICollectionView(frame: .zero, collectionViewLayout: UICollectionViewFlowLayout())
+    collectionView.showsHorizontalScrollIndicator = false
```

Comment:
I vaguely recall that there's a way to disable scrolling completely on a collection view (or scroll view?) and that wouldn't be necessary then.